### PR TITLE
sidebar functionality: collapsible provider-types + scroll to provider when icon clicked on map

### PR DESCRIPTION
**Pull request by bhinebaugh**
functionality was in prior (separate) branches and PRs; this just makes them work together, namely expands a provider type before attempting to scroll to provider within that type if necessary

### CODE REVIEW

File: src/components/ProviderList/provider-list.js [hunk 1]
```diff
@@ -9,21 +9,47 @@ class ProviderList extends Component {
     super(props);
     this.listElementRef = React.createRef();
     this.lastHighlightedRef = React.createRef();
+    this.state = {
+      collapsedProviderTypes: [],
+    }
+  }
+
+  toggleProviderType = (id) => {
+    console.log('toggling provider type', id);
```
Comment (alexjball) [inline src/components/ProviderList/provider-list.js:18]: nit: remove line

File: src/components/ProviderList/provider-list.js [hunk 2]
```diff
@@ -9,21 +9,47 @@ class ProviderList extends Component {
...
+        let providerTypeProviders = providerType.providers.map(p => p.id);
+        let typeIsCollapsed = this.state.collapsedProviderTypes.includes(providerType.id);
+        if (typeIsCollapsed && providerTypeProviders.includes(nowHighlighted[0])) { this.toggleProviderType(providerType.id); }
+      }
+
+      // add a small delay when updating scrollTop to avoid edge case 
+      // of 'open' height not being taken into account 
+      // when scrolling the list upward
+      let timeoutID = setTimeout( () => {
+        this.listElementRef.current.scrollTop = this.lastHighlightedRef.current.offsetTop
```
Comment (alexjball) [inline src/components/ProviderList/provider-list.js]: I remember there was an issue where lastHighlightedRef could be missing if we try to scroll when the category is collapsed. Is that still an issue?

File: src/components/MenuDropdown/index.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
 import MenuDropdown from "./menu-dropdown";
+import CollapsibleMenuDropdown from "./menu-dropdown-collapsible";
 
-export default MenuDropdown;
+export { CollapsibleMenuDropdown, MenuDropdown };
```
Comment (alexjball) [inline src/components/MenuDropdown/index.js]: It looks like MenuDropdown is unused now. Can we remove it?

File: src/components/ProviderList/provider-list.js [hunk 4]
```diff
@@ -9,21 +9,47 @@ class ProviderList extends Component {
...
-    let newhlp = this.props.highlightedProviders;
-    if (newhlp.length && newhlp[0] !== previousProps.highlightedProviders[0]) {
-      // CSS 'scroll-behavior: smooth' animates the scroll when scrollTop is updated;
-      // adding a delay avoids edge case of scroll-upward not taking 'open' height into account
-      setTimeout(
-        () =>
-          (this.listElementRef.current.scrollTop = this.lastHighlightedRef.current.offsetTop),
+    let nowHighlighted = this.props.highlightedProviders;
+    
+    if (nowHighlighted.length && nowHighlighted[0] !== previousProps.highlightedProviders[0]) {
```
Comment (alexjball) [inline src/components/ProviderList/provider-list.js]: WDYT about changing this line to 
```
    if (nowHighlighted.length && !previousProps.highlightedProviders.includes(nowHighlighted[0])) {
```
This way, we only scroll if nowHighlighted[0] is newly highlighted, so deselecting a provider doesn't scroll.

File: src/components/MenuDropdown/menu-dropdown.css [hunk 5]
```diff
@@ -0,0 +1,12 @@
+.dropdown-menu {
+    width: 100%;
```
Comment (alexjball) [inline src/components/MenuDropdown/menu-dropdown.css:2]: Consider adding a few pixels of padding-top and padding-bottom to this rule, I think it looks a bit crowded when multiple categories are collapsed
collapsible-dd

**Review by alexjball — COMMENTED** [review]
Looks great! Maybe we can talk through my comment about lastHighlightedRef tonight.

Also Travis is complaining about unused variables. These show up as warnings in the development console when you load the page.

File: src/components/MenuDropdown/index.js (hunk 3 above)
Comment (bhinebaugh) [reply src/components/MenuDropdown/index.js]: I refactored MenuDropdown and removed CollapsibleMenuDropdown instead

File: src/components/MenuDropdown/menu-dropdown.css (hunk 5 above)
Comment (bhinebaugh) [reply src/components/MenuDropdown/menu-dropdown.css:2]: definitely did. I added some padding for any .dropdown-menu that is also .closed

File: src/components/MenuDropdown/index.js (hunk 3 above)
Comment (bhinebaugh) [reply src/components/MenuDropdown/index.js]: MenuDropdown now supports a prop `collapsible` that enables that functionality

File: src/components/ProviderList/provider-list.js (hunk 2 above)
Comment (alexjball) [reply src/components/ProviderList/provider-list.js]: Resolved offline. There was an issue stemming from the deselect-then-scroll bug, but that's fixed in this PR.

**Pull request merged**